R/utilities.R

```diff
@@ -805,181 +805,414 @@ trim_scale_expression = function(x){
-                                                   mode = "slms-3"){
-
-  max_region = 1000000
-  if(missing(metadata)){
-    metadata = get_gambl_metadata()
-  }
-  if(missing(this_region)){
-    this_region = paste0(chromosome, ":", start_pos, "-", end_pos)
-  }else{
-    chunks = region_to_chunks(this_region)
-    chromosome = chunks$chromosome
-    start_pos = as.numeric(chunks$start)
-    end_pos = as.numeric(chunks$end)
-  }
-  region_size = end_pos - start_pos
-  if(region_size < max_region){
-    message(paste("processing bins of size", window_size, "across", region_size, "bp region"))
-  }else{
+
+calc_mutation_frequency_sliding_windows <- function(this_region,
+                                          chromosome,
+                                          start_pos,
+                                          end_pos,
+                                          these_samples_metadata,
+                                          these_sample_ids,
+                                          maf = NULL,
+                                          projection = "grch37",
+                                          slide_by = 100,
+                                          window_size = 1000,
+                                          return_format = "long",
+                                          min_count_per_bin = 0,
+                                          return_count = TRUE,
+                                          drop_unmutated = FALSE,
+                                          from_indexed_flatfile = TRUE,
+                                          mode = "slms-3") {
+  # Create objects to describe region both as string and individual objects
+  try(if (missing(this_region) & missing(chromosome)) {
+    stop("No region information provided. Please provide a region as a string in the chrom:start-end format, or as individual arguments. ")
+  })
+
+  if ((drop_unmutated | min_count_per_bin > 0) & return_format == "wide") {
+    message("To return a wide table, all samples and windows must be kept. Ignoring drop_unmutated and min_count_per_bin arguments. ")
+  }
+
+  if (missing(this_region)) {
+    this_region <- paste0(
+      chromosome, ":", start_pos, "-",
+      end_pos
+    )
+  } else {
+    chunks <- GAMBLR:::region_to_chunks(this_region)
+    chromosome <- chunks$chromosome
+    start_pos <- as.numeric(chunks$start)
+    end_pos <- as.numeric(chunks$end)
+  }
+
+  # Check for provided metadata, else use GAMBL metadata
+  if (missing(these_samples_metadata)) {
+    metadata <- get_gambl_metadata(
+      seq_type_filter = c("capture", "genome"),
+      tissue_status_filter = "tumour"
+    )
+  } else {
+    metadata <- these_samples_metadata
+  }
+  # Ensure metadata is subset to specified sample IDs
+  if (!missing(these_sample_ids)) {
+    metadata <- dplyr::filter(
+      metadata,
+      sample_id %in% these_sample_ids
+    )
+  } else {
+    these_sample_ids <- metadata$sample_id
+  }
+
```

Comment:
How do we deal with values in these_sample_ids that are not in the metadata? Should we catch those and warn the user?